src/io/GammaWriter.cpp

```diff
@@ -5,103 +5,119 @@
+		_globalLength[d] = domain->getGlobalLength(d);
+	}
+	_numComp = domain->getNumberOfComponents() + 1;  // 0 stands for all components
+
+	_gamma.resize(_numComp);
+
+	// Rank 0 writes data to file
+	if (domainDecomp->getRank() == 0) {
+		string resultfilename(_outputPrefix + ".dat");
 		_gammaStream.open(resultfilename);
 		_gammaStream.precision(6);
-		const auto now = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
-		tm unused{};
-		_gammaStream << "# mardyn MD simulation starting at " << std::put_time(localtime_r(&now, &unused), "%c") << endl;
-		_gammaStream << "#\tgamma" << endl;
-	}
+		_gammaStream << setw(24) << "simstep";
+        for (unsigned int componentId = 0; componentId < _numComp; ++componentId) {
+            _gammaStream << setw(22) << "gamma[" << componentId << "]";
+        }
+        _gammaStream << std::endl;
+        _gammaStream.close();
+    }
 }
 
 void GammaWriter::endStep(ParticleContainer *particleContainer, DomainDecompBase *domainDecomp, Domain *domain,
                           unsigned long simstep)
 {
-	calculateGamma(particleContainer, domainDecomp);
-	if((domainDecomp->getRank() == 0) && (simstep % _writeFrequency == 0)){
-		double globalLength[3];
-		for(int d = 0; d < 3; ++d) {
-			globalLength[d] = domain->getGlobalLength(d);
-		}
-		_gammaStream << simstep;
-		for(unsigned int componentId = 0; componentId < domain->getNumberOfComponents(); ++componentId){
-			_gammaStream << "\t" << getGamma(componentId, globalLength)/_writeFrequency;
+    calculateGamma(particleContainer, domainDecomp);
+
+    // Write every _writeFrequency steps; do not write data directly after (re)start
+    if ((simstep % _writeFrequency == 0) && (simstep > global_simulation->getNumInitTimesteps())) {
+		// Rank 0 writes data to file
+		if (domainDecomp->getRank() == 0) {
+			string resultfilename(_outputPrefix + ".dat");
+			
+			_gammaStream.open(resultfilename, std::ios::app);
+			_gammaStream << FORMAT_SCI_MAX_DIGITS << simstep;
+			for (unsigned int componentId = 0; componentId < _numComp; ++componentId) {
+				_gammaStream << FORMAT_SCI_MAX_DIGITS << getGamma(componentId)/_writeFrequency;
+			}
+			_gammaStream << std::endl;
+			_gammaStream.close();
 		}
-		_gammaStream << endl;
-		resetGamma();
-	}
+        resetGamma(domain);
+    }
 }
 
-void GammaWriter::finish(ParticleContainer * /*particleContainer*/,
-						 DomainDecompBase * /*domainDecomp*/, Domain * /*domain*/){
-	_gammaStream.close();
+inline void GammaWriter::resetGamma(Domain *domain) {
+    for (unsigned componentId = 0; componentId < _numComp; ++componentId) {
+        _gamma.at(componentId) = 0.0;
+    }
 }
 
-void GammaWriter::resetGamma() {
-	for(unsigned componentId = 0; componentId < _simulation.getEnsemble()->getComponents()->size(); ++componentId) {
-		_Gamma[componentId] = 0;
-	}
+inline double GammaWriter::getGamma(unsigned id) {
+    return (_gamma.at(id)/(_globalLength.at(0)*_globalLength.at(2)*_numInterfaces));
```

Comment:
`.at()` checks for bounds automatically, while `[]` is faster. I thought, that it would be better to be safe by using `.at()`. But if everything is implemented correctly (*fingers crossed*), maybe `[]` would be preferred.

Reply:
your call. If this stuff is speed crititcal go for `[]`. If you messed up you should get segfaults anyway ;)